### PR TITLE
bugfix: Solve the problem of outputting a large amount of logs after edgecore exits

### PR DESCRIPTION
**Pull request by kadisi**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
> /kind feature

 /kind bug
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Bugfix: Remove redundant logs when EdgeCore exits
```

### CODE REVIEW

File: edge/pkg/devicetwin/devicetwin.go [hunk 1]
```diff
@@ -1,17 +1,20 @@
 package devicetwin
 
 import (
+	"context"
+
 	"k8s.io/klog"
 
 	"github.com/kubeedge/beehive/pkg/core"
-	"github.com/kubeedge/beehive/pkg/core/context"
+	beehivecontext "github.com/kubeedge/beehive/pkg/core/context"
```
Comment (fisherxu) [inline edge/pkg/devicetwin/devicetwin.go]: `beehiveContext `

**Comment by kevin-wangzefeng** [comment]
Revised release note to: "Bugfix: Remove redundant logs when EdgeCore exits"
Let me know when it's ready for final review.

File: edge/pkg/devicetwin/devicetwin.go (hunk 1 above)
Comment (kadisi) [reply edge/pkg/devicetwin/devicetwin.go]: done

**Comment by kadisi** [comment]
@kevin-wangzefeng @fisherxu PTAL

File: edge/pkg/edged/edged.go [hunk 2]
```diff
@@ -343,15 +347,12 @@ func (e *edged) Start(c *context.Context) {
 	go e.pluginManager.Run(edgedutil.NewSourcesReady(), utilwait.NeverStop)
 
 	klog.Infof("starting syncPod")
-	e.syncPod()
+	e.syncPod(ctx)
 }
 
 func (e *edged) Cleanup() {
-	if err := recover(); err != nil {
```
Comment (fisherxu) [inline edge/pkg/edged/edged.go]: +1, no need recover here.

File: edge/pkg/edgehub/module.go [hunk 3]
```diff
@@ -35,12 +51,63 @@ func (eh *EdgeHub) Group() string {
...
+		}
+		// execute hook func after connect
+		eh.pubConnectInfo(true)
+		go eh.routeToEdge(ctx)
+		go eh.routeToCloud(ctx)
+		go eh.keepalive(ctx)
+
+		// wait the stop singal
+		// stop authinfo manager/websocket connection
+		<-eh.retryChan
```
Comment (fisherxu) [inline edge/pkg/edgehub/module.go]: reconnectChan？

File: edge/pkg/devicetwin/devicetwin.go [hunk 4]
```diff
@@ -33,21 +40,24 @@ func (dt *DeviceTwin) Group() string {
...
-	dt.dtcontroller = controller
-	dt.context = c
-	err = controller.Start()
+func (dt *DeviceTwin) Start(c *beehiveContext.Context) {
+	var ctx context.Context
+	dtContexts, _ := dtcontext.InitDTContext(c)
+	dt.HeartBeatToModule = make(map[string]chan interface{})
+	dt.DTModules = make(map[string]dtmodule.DTModule)
+	dt.DTContexts = dtContexts
+	dt.Context = c
```
[the remark this answers is not in the document]
Comment (kadisi) [reply edge/pkg/devicetwin/devicetwin.go]: ok

File: edge/pkg/edgehub/module.go (hunk 3 above)
Comment (kadisi) [reply edge/pkg/edgehub/module.go]: ok

File: edge/pkg/devicetwin/dtcontroller.go [hunk 5]
```diff
@@ -341,3 +270,56 @@ func classifyMsg(message *dttype.DTMessage) bool {
 	}
 	return false
 }
+
+func (dt *DeviceTwin) mainLoop(ctx context.Context) {
```
[the remark this answers is not in the document]
Comment (kadisi) [reply edge/pkg/devicetwin/dtcontroller.go]: ok

**Pull request merged**